cdap-watchdog/src/main/java/co/cask/cdap/logging/gateway/handlers/LogHandler.java

```diff
@@ -82,6 +84,16 @@ public void getLogs(HttpRequest request, HttpResponder responder, @PathParam("na
     doGetLogs(responder, loggingContext, fromTimeSecsParam, toTimeSecsParam, escape, filterStr, null);
   }
 
+  private LoggingContext getLoggingContext(String namespaceId, String appId, ProgramType programType, String programId,
+                                           String runId, Map<String, String> systemArgs) {
+    if (systemArgs != null && systemArgs.containsKey("workflowRunId")) {
+      String workflowRunId = systemArgs.get("workflowRunId");
+      String workflowId = systemArgs.get("workflowName");
+      return new WorkflowProgramLoggingContext(namespaceId, appId, workflowId, workflowRunId, programType, programId);
```

Comment:
Let's move this logic to `LoggingContextHelper.getLoggingContextWithRunId()`, so that we have one place where we generate `LoggingContext`.